Zombie states

taramic42 opened the pull request:
## What issue is this PR solving?
Closes: https://github.com/StudioTBA/CoronaIO/issues/58
Closes: https://github.com/StudioTBA/CoronaIO/issues/57
Closes: https://github.com/StudioTBA/CoronaIO/issues/56

## Description of change
Zombie hordes now have multiple states that they can use to influence their behavior

Idle (where they do nothing)
Wander
Attack (where they head towards wherever the closest human is within a certain range)
SeekClosestHuman
Flee

Several booleans were added to FlockManager (always_flee, fixate_on_target, attack_if_able and stop)
that the player can toggle with the keys Z X C and V to influence the behavior of each individual horde. These behaviors are maintained even when horde is not being actively controlled

## How to reproduce the changes
Open Sandbox scene
Watch FlockCenter wander around
Press V to stop it
Press Z to make it flee from human if it gets too close
Press X to force it to attack if within range of a human
Press C to make FlockCenter fixate on closest human/current target and ignore all others
Modify flee_distance_goal attribute in Zombie_Flee component if you want Horde to run more/less from human

IrregularMeter committed:
Add MiniMap Folder and MapTexture Script
  - MapTexture Script takes a screenshot of the Environment

IrregularMeter committed:
Set map texture to Canvas

IrregularMeter committed:
Add miniMap Icon to follow GameObject

IrregularMeter committed:
Add Unity UI Extensions Package

IrregularMeter committed:
Add ViewPortBox in MiniMap

IrregularMeter committed:
Save scene

IrregularMeter committed:
Add CameraClickAndGo Script / Needs Refactor

IrregularMeter committed:
Add MiniMapManager and Helper / Refactor

IrregularMeter committed:
Refactor

IrregularMeter committed:
Add MouseButton Check for Camera Movement in Minimap

IrregularMeter committed:
Add MiniMapManager Prefab
  - Add MiniMap in SandBox Scene

IrregularMeter committed:
Adjust map pic size

IrregularMeter committed:
Add the following changes:
  - CalculateCenterOfMass script on the CenterOfMass object (Child of FlockCenter)
  - Add Scripts for Click and Focus on Horde
  - Add ZombieHordeBlipManager (Takes care of Deletion and Position Update)
  - Add ZombieHordeBlipCreator script

IrregularMeter committed:
Refactor

IrregularMeter committed:
Refactor

IrregularMeter committed:
Add Select and Deselect of Horde on Minimap

IrregularMeter committed:
Add Camera lock to Horde

IrregularMeter committed:
Add Outline Shader to Zombies

gnoriega27 committed:
rebase with master

gnoriega27 committed:
Duplicate

gnoriega27 committed:
fixed merge

IrregularMeter committed:
Add ZombieHordeBlipCreator script to FlockCenter prefab

IrregularMeter committed:
Add MiniMap to Game Scene

taramic42 committed:
Trying to get state machine started

taramic42 committed:
Wander and Idle states switch between each other depending on if horde is active or not

taramic42 committed:
Added transition logic for Zombie_Attack, Zombie_Wander and Idle_Zombie

taramic42 committed:
Fixed some wonky movement in Zombie_Wander

taramic42 committed:
Merge branch 'master' of github.com:StudioTBA/CoronaIO into Zombie_States

# Conflicts:
#	Assets/Prefabs/Zombie/FlockCenter.prefab
#	Assets/Scenes/Sandbox.unity

taramic42 committed:
Implemented controls for horde behaviour allowing player to set individual horde strategies (AlwaysFlee, Attack/Rampage, Fixate, Stop). Implemented Zombie_Flee state

mjimenez98 reviewed (dismissed):
I like this a lot. Nice work 💯 

I won't comment on values related to game balancing so I think it is awesome as it is.

mjimenez98 commented:
I just noticed something else. When splitting a horde, the new one goes straight to wander. I would suggest making it inherit the properties of the original (like a clone). What do you think?

IrregularMeter committed:
Delete Examples Folder from Unity UI Package

IrregularMeter committed:
Fix MiniMap Bug where trying to lock on another horde while being locked onto one initially didn't work

IrregularMeter committed:
Add free movement option when locked to horde
  - Add feature so that you only DeSelect the horde when you left click on the minimap

IrregularMeter committed:
Save scene

taramic42 committed:
Removed fixated_on_target. Made split hordes copy states from original horde

IrregularMeter committed:
Trying to fix merging of hordes

taramic42 committed:
Futsed around with horde splitting to make sure it highlights zombies properly

IrregularMeter committed:
Add Singleton For ZombieHordeBlips

IrregularMeter committed:
Apply MiniMapManager prefab change

taramic42 committed:
Merge branch 'lyonelz/MiniMap' of github.com:StudioTBA/CoronaIO into Zombie_States

# Conflicts:
#	Assets/MiniMap/MapPic.png
#	Assets/Scripts/Flock/HordeOrganizer.cs

taramic42 committed:
KILLED THE BUG (Credit goes to Lyonel)

IrregularMeter committed:
Remove Debug.Log() on ClickAndGoToHorde
